lib/tty/reader/console.rb

```diff
@@ -45,7 +45,17 @@ def initialize(input)
       def get_char(options)
         mode.raw(options[:raw]) do
           mode.echo(options[:echo]) do
-            if options[:nonblock]
+            if options[:detect_esc]
+              begin
```

Comment:
No reason to have this code. This method is called recursively both on unix and windows. I appreciate intention behind it but that's exactly what the rest of `get_codes` does.